### ISSUE TITLE
Idea: Git-backed Profiles

### ISSUE THREAD

**Issue by jtopjian** [jtopjian/waffles#83]
Currently, Waffles does not support modules, or the ability to make configuration scripts so generalized that they can be used by a large audience. However, I've always said that profiles and scripts can be re-used _internally_. This includes common scripts such as setting an apt proxy, adding standard users, etc.

Having Waffles be able to pull a script from a git repo has a few benefits:
- The script can be versioned and locked down to a single version
- Variations of the script could be branched for dev, prod, etc
- Basically all the benefits of storing code in a repository

However, scripts are part of profiles and profiles can also include static files. How could Waffles easily handle this?

**Comment by jtopjian** [comment]
#93 adds this support
